tv_apps/smart-home/js/card_picker.js

```diff
@@ -38,16 +41,69 @@
       this.container.addEventListener('keyup', this.onKeyUp.bind(this), true);
 
       this._selectedElements = this.gridView.getElementsByClassName('selected');
+
+      this.input.addEventListener('focus', () => {
+        setTimeout(() => {
+          this.input.setSelectionRange(
+            this.input.value.length, this.input.value.length);
+        }, 0);
+      });
+    },
+
+    openKeyboard: function () {
+      this.panel.classList.add('hidden');
+      this.input.readOnly = false;
+      this.input.blur();
+      this.input.focus();
+      // There is a case that
+      // when user long presses Space key on virtual keyboard,
+      // it would hide keyboard and blur input.
+      // In this case, the focused element inside the spatialNavigator would
+      // still be the input element and the input element's readOnly
+      // would still be false.
+      // So we have to call closeKeyboard to make all the states back to right.
+      var handleBlur = () => {
+        this.input.removeEventListener('blur', handleBlur);
+        var focus = this._spatialNavigator.getFocusedElement();
+        if (focus === this.input && this.isKeyboardOpened) {
```

Comment:
I guess we don't need to check the status here since we should always call `closeKeyboard` whenever the `input` loses the focus no matter what causes it. Right?